### PR TITLE
Re-Implementation

### PR DESCRIPTION
**Pull request by EliranBarDavid**
Fixing Connection issue
Adding support for UIELementAction and BrowserAction to Mobile.
Enhanced mobile action
split drivers per types

### CODE REVIEW

File: Ginger/GingerCore/Drivers/Mobile/Perfecto/PerfectoDriver.cs [hunk 1]
```diff
@@ -302,101 +402,179 @@ public enum eSwipeSide {
...
 
-        public override bool IsRunning() {
+        public override bool IsRunning()
+        {
             return ConnectedToDevice;
         }
 
-        private string GetKeyName(string skey) {
-            switch (skey) {
+        private string GetKeyName(string skey)
```
Comment (MeniKadosh1) [inline Ginger/GingerCore/Drivers/Mobile/Perfecto/PerfectoDriver.cs]: do cleanup for methods needed only for GenElement action

File: Ginger/GingerCore/Platforms/Agent.cs [hunk 2]
```diff
@@ -114,8 +114,14 @@ public enum eDriverType
             MobileAppiumAndroidBrowser,
             [Description("Mobile Appium IOS Browser")]
             MobileAppiumIOSBrowser,
-            [Description("Perfecto Mobile")]
-            PerfectoMobile,
+            [Description("Perfecto Mobile Android Device")]
```
Comment (MeniKadosh1) [inline Ginger/GingerCore/Platforms/Agent.cs]: change to be same terms as used in Appuim

File: Ginger/GingerCore/Drivers/Mobile/Perfecto/PerfectoDriver.cs [hunk 3]
```diff
@@ -56,148 +66,226 @@ public class PerfectoDriver : DriverBase {
...
+        [UserConfiguredDescription("Implicit Wait for Web Action Completion")]
+        public int ImplicitWait { get; set; }
+
         public bool ConnectedToDevice = false;
-        private RemoteWebDriverExtended Driver;
-        
-        public enum eSwipeSide {
-            Up, Down, Left, Right
+        //private AppiumDriver<IWebElement> AppiumDriver;
+        private RemoteWebDriver AppiumDriver;
```
[the remark this answers is not in the document]
Comment (EliranBarDavid) [reply Ginger/GingerCore/Drivers/Mobile/Perfecto/PerfectoDriver.cs]: PerfectoDriver is the name of the class so i called it mDriver.

**Comment by MeniKadosh1** [comment]
PR was merged and branch was deleted

**Pull request merged**